proctoru/proctoru.py

```diff
@@ -188,6 +188,96 @@ def get_block_id(self):
         """
```

Comment:
This is good but, how are we sure the whole refactor works the same as the old code ?